### PR TITLE
XOR-303 [FIX] Ensure Slider field max, min and step validation and changes

### PR DESCRIPTION
**Pull request by NikitaJadhavXoriant**
### Product areas affected

Widgets -> Add Form -> Advanced Section -> Select Slider field 

### What does this PR do?

Implemented code to add validation for Slider field in form

### JIRA ticket

[JIRA](https://weboo.atlassian.net/browse/XOR-303)

### Result

https://user-images.githubusercontent.com/108272606/205971374-b240668a-7679-462a-9cc8-31c133ce4600.mp4

### Checklist

None

### Deployment instructions

None

### Author concerns

None

### CODE REVIEW

**Review by yogeshmsharma-xoriant — APPROVED** [review]
looks ok

File: templates/configurations/form.interface.hbs [hunk 1]
```diff
@@ -96,21 +96,23 @@
     </div>
 
     {{#isSlider}}
-    <div class="form-group cleafix">
-      <div class="col-xs-12">
-        <label>Minimum Value Number</label>
+    <div class="form-group cleafix" :class="{ 'has-error': _fieldSliderMaxMinError || _fieldSliderStepError}">
+      <div class="col-xs-12 slider-margin-bottom" >
```
Comment (tonytlwu) [inline templates/configurations/form.interface.hbs:100]: This is a terrible solution. Looking at the HTML markup, it's obvious that the problem with the padding occurs due to incorrectly usage of the Bootstrap framework or adherence to how we have used it in the rest of this file.

Every configuration field should be wrapped in `.form-group`, which provides the spacing between each configuration field. You've stuck 3 configurations into one `.form-group` here. Please read up on the rest of this file and follow our convention.

File: templates/configurations/form.interface.hbs [hunk 2]
```diff
@@ -96,21 +96,23 @@
     </div>
 
     {{#isSlider}}
-    <div class="form-group cleafix">
-      <div class="col-xs-12">
-        <label>Minimum Value Number</label>
+    <div class="form-group cleafix" :class="{ 'has-error': _fieldSliderMaxMinError || _fieldSliderStepError}">
```
Comment (tonytlwu) [inline templates/configurations/form.interface.hbs:99]:
```suggestion
    <div class="form-group cleafix" :class="{ 'has-error': _fieldSliderMaxMinError || _fieldSliderStepError }">
```

File: js/libs/core.js [hunk 3]
```diff
@@ -440,6 +446,30 @@ Fliplet.FormBuilder = (function() {
...
+
+        if (this._componentName === 'flSlider') {
+          if (this.min >= this.max) {
+            return 'The minimum value cannot be higher than the maximum value';
+          }
+        }
+      };
+
+      component.computed._fieldSliderStepError = function() {
+        if (!this.min && !this.max && !this.step) {
```
Comment (tonytlwu) [inline js/libs/core.js:462]: Use early exit. @yogeshmsharma-xoriant Does your team have difficult understanding what the early exit coding practice means?

```suggestion
        if (this._componentName !== 'flSlider' || !this.min && !this.max && !this.step) {
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: templates/configurations/form.interface.hbs (hunk 1 above)
Comment (NikitaJadhavXoriant) [reply templates/configurations/form.interface.hbs]: done the changes

File: templates/configurations/form.interface.hbs (hunk 2 above)
Comment (NikitaJadhavXoriant) [reply templates/configurations/form.interface.hbs]: done

File: js/libs/core.js (hunk 3 above)
Comment (NikitaJadhavXoriant) [reply js/libs/core.js]: added early exit.

**Review by yogeshmsharma-xoriant — APPROVED** [review]
looks ok to me

File: css/form.css [hunk 4]
```diff
@@ -1054,10 +1054,6 @@ html[dir="rtl"] .radio.radio-icon input[type="radio"] + label > span.check {
   color: #ffffff;
 }
 
-[data-widget-package="com.fliplet.form-builder"] .form-group.fl-range-slider.readonly .range-slider-handle-value::before {
```
Comment (tonytlwu) [inline css/form.css:1057]: Why is this removed if we're only adding validation to the configurations?

File: js/libs/core.js [hunk 5]
```diff
@@ -354,6 +354,25 @@ Fliplet.FormBuilder = (function() {
           }
         }
 
+        if (this._componentName === 'flSlider') {
+          if (!this.step) {
+            data.step = 1;
+          }
+
+
```
Comment (tonytlwu) [inline js/libs/core.js:362]:
```suggestion
```

File: js/libs/core.js [hunk 6]
```diff
@@ -354,6 +354,25 @@ Fliplet.FormBuilder = (function() {
...
+          if (!this.min) {
+            data.min = 0;
+          }
+
+          if (!this.max) {
+            data.max = 100;
+          }
+
+          if ((Number(data.max) - Number(data.min)) % Number(data.step) != 0) {
```
Comment (tonytlwu) [inline js/libs/core.js:371]: Seems less messy if you just set `data.max = Number(data.max)` all before checking the values so that they are all numbers.

Comment (tonytlwu) [inline js/libs/core.js:371]: @yogeshmsharma-xoriant Can you explain why this didn't get picked up by ESLint?

```suggestion
          if ((Number(data.max) - Number(data.min)) % Number(data.step) !== 0) {
```

File: js/libs/core.js [hunk 7]
```diff
@@ -440,6 +459,26 @@ Fliplet.FormBuilder = (function() {
         return '';
       };
 
+      component.computed._fieldSliderMaxMinError = function() {
+        if (this._componentName !== 'flSlider' || !this.min && !this.max) {
+          return;
+        }
+
+        if (Number(this.min) >= Number(this.max)) {
+          return 'The minimum value cannot be higher than the maximum value';
```
Comment (tonytlwu) [inline js/libs/core.js:468]:
```suggestion
          return 'The maximum value must be higher than the minimum value';
```

File: templates/configurations/form.interface.hbs [hunk 8]
```diff
@@ -96,21 +96,27 @@
     </div>
 
     {{#isSlider}}
-    <div class="form-group cleafix">
-      <div class="col-xs-12">
-        <label>Minimum Value Number</label>
+    <div class="form-group cleafix" :class="{ 'has-error': _fieldSliderMaxMinError || _fieldSliderStepError }">
+      <div class="col-xs-12" >
```
Comment (tonytlwu) [inline templates/configurations/form.interface.hbs:100]:
```suggestion
      <div class="col-xs-12">
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: css/form.css (hunk 4 above)
Comment (yogeshmsharma-xoriant) [reply css/form.css:1057]: removed un-necessary css which was already present in ui.rangeslider.css.

File: js/libs/core.js (hunk 6 above)
Comment (yogeshmsharma-xoriant) [reply js/libs/core.js:371]: seems linter is not getting applied for fliplet-widget-form-builder\libs\*,*.

**Review by yogeshmsharma-xoriant — APPROVED** [review]
looks ok to me after my  feedback today

File: js/libs/core.js (hunk 5 above)
Comment (NikitaJadhavXoriant) [reply js/libs/core.js:361]: done

File: js/libs/core.js (hunk 6 above)
Comment (NikitaJadhavXoriant) [reply js/libs/core.js]: done

File: js/libs/core.js (hunk 7 above)
Comment (NikitaJadhavXoriant) [reply js/libs/core.js]: done

File: templates/configurations/form.interface.hbs (hunk 8 above)
Comment (NikitaJadhavXoriant) [reply templates/configurations/form.interface.hbs]: done

**Review by yogeshmsharma-xoriant — DISMISSED** [review]
looks ok now

**Review by tonytlwu — APPROVED** [review]

File: js/libs/core.js (hunk 6 above)
Comment (tonytlwu) [reply js/libs/core.js]: @yogeshmsharma-xoriant Can you make sure it is from now on please

File: css/form.css (hunk 4 above)
Comment (tonytlwu) [reply css/form.css:1057]: @yogeshmsharma-xoriant Based on [this PR rejection](https://github.com/Fliplet/fliplet-api/pull/5599#pullrequestreview-1220772074), does this need to be added back?

File: js/libs/core.js (hunk 6 above)
Comment (yogeshmsharma-xoriant) [reply js/libs/core.js]: sure @tonytlwu. I didn't do any search on why it is not working in lib\***.  Will avoid it

**Review by yogeshmsharma-xoriant — DISMISSED** [review]
looks ok

**Review by yogeshmsharma-xoriant — APPROVED** [review]
looks ok
